source/common/upstream/upstream_impl.cc

```diff
@@ -932,14 +938,14 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(const HostVector& new_hosts,
   // avoid pivoting on this entirely and probably just force a host set refresh if any weights
   // change.
   info_->stats().max_host_weight_.set(max_host_weight);
-
   if (!hosts_added.empty() || !current_hosts.empty()) {
     hosts_removed = std::move(current_hosts);
     current_hosts = std::move(final_hosts);
     return true;
   } else {
-    // During the search we moved all of the hosts from hosts_ into final_hosts so just
-    // move them back.
+    // During the search populated final_hosts with either from hosts_ or from all_hosts, so
```

Comment:
This sentence is a little hard to read and I think the comment is important. Can you tighten?